### PR TITLE
Extract Kotlin name from KClass instead of Java name.

### PR DESCRIPTION
**Pull request by JakeWharton**
Refs #86. Not the whole way there, but a first step.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Rebased

File: src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt [hunk 1]
```diff
@@ -137,12 +137,12 @@ class ClassNameTest {
     } catch (expected: IllegalArgumentException) {
     }
 
-    try {
-      ClassName.get(Array<Any>::class)
-      fail()
-    } catch (expected: IllegalArgumentException) {
-    }
-
+    // TODO
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt]: ?

File: src/test/java/com/squareup/kotlinpoet/TypesEclipseTest.kt [hunk 2]
```diff
@@ -36,6 +37,7 @@ import javax.lang.model.util.Types
 import javax.tools.DiagnosticCollector
 import javax.tools.JavaFileObject
 
+@Ignore("Not clear this test is useful to retain in the Kotlin world")
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/TypesEclipseTest.kt]: Yeah.

File: src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt (hunk 1 above)
Comment (JakeWharton) [inline src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt]: See https://github.com/square/kotlinpoet/issues/86#issuecomment-305071978. This is the language leaking implementation details of the JVM 😢 .

I'm not sure what to do here yet...

**Pull request merged**